### ISSUE TITLE
trainer.validate throws error when using ckpt_path

### ISSUE THREAD

**Issue by zmurez** [PyTorchLightning/pytorch-lightning#7600]
## 🐛 Bug: trainer.validate throws error when using ckpt_path
When using `trainer.validate()` with `ckpt_path=` before running `trainer.fit()` I get the following error:
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-4-b4d5f405dd04> in <module>()
      1 # now this doesn't work
----> 2 trainer.validate(ckpt_path='lightning_logs/version_0/checkpoints/epoch=2-step=938.ckpt', val_dataloaders=val)

2 frames
/usr/local/lib/python3.7/dist-packages/pytorch_lightning/trainer/connectors/data_connector.py in attach_dataloaders(self, model, train_dataloader, val_dataloaders, test_dataloaders, predict_dataloaders)
    103 
    104         if val_dataloaders is not None:
--> 105             model.val_dataloader = _PatchDataLoader(val_dataloaders)
    106 
    107         if test_dataloaders is not None:

AttributeError: 'NoneType' object has no attribute 'val_dataloader'
```

Note that the error can also be resolved by running `trainer.validate()` with `model=` before running it again with `ckpt_path=`.

Seems like the model is not being initialized properly when passing `ckpt_path`.

## To Reproduce
https://colab.research.google.com/drive/1902bZYoUuGbWWH_4J2l9jb-cP2W_Xp2M?usp=sharing

Thanks!

**Comment by i-aki-y** [comment]
From a quick investigation, validate method seems to use `trainer.lightning_module` to get the model information if the model arg is not provided. But just after the initialization, the `trainer.lightning_module` is `None`value.

I guess that the `validate` method does not support the use case for calling `validate(checkpoint=ckpt_path)` without the `fit` method if the model argument is not provided.

**Comment by zmurez** [comment]
I agree with your analysis. However I think this is an important use case because one often wants to re-run validation on a model that was trained in the past (in fact that seems to be the main reason for adding `validate` in the first place). Without this I don't see what the intended use of `ckpt_path` is?

It might also be nice if you could pass `model` instead of `ckpt_path` and then set `trainer.resume_from_checkpoint`. However this doesn't work (validate runs but the checkpoint is not loaded).

What is the suggested way validate an old model? I guess the simplest is to use `model=LightningModule.load_from_checkpoint()` and then pass `model` to validate.

**Comment by i-aki-y** [comment]
> What is the suggested way validate an old model? I guess the simplest is to use model=LightningModule.load_from_checkpoint() and then pass model to validate.

Yes, I think this is one of the simplest ways.
Anyway, if this is not a bug, I think this should be documented somewhere.

**Comment by tchaton** [comment]
@carmocca ?

**Comment by ananthsub** [comment]
https://github.com/PyTorchLightning/pytorch-lightning/issues/6030 seems like the root cause. I think it should be valid behavior to provide a model checkpoint to `trainer.validate` w/o having previously run `trainer.fit/validate/test/predict`

**Comment by tchaton** [comment]
Dear @zmurez,

IMO, I think it is an mis-usage of the Trainer.validate function. The error is triggered as the trainer doesn't have a reference to the model. Therefore, you should provide the model the validate function as Lightning just save the state_dict, not the actual model class.

This should work:

```
trainer.validate(model, ckpt_path='lightning_logs/version_0/checkpoints/epoch=2-step=938.ckpt', val_dataloaders=val)
```

**Comment by carmocca** [comment]
> This should work:

I agree. Mentioned here: https://github.com/PyTorchLightning/pytorch-lightning/pull/7605#discussion_r635668438.
@yifuwang said he would take it.

> I think we should raise a MisConfigurationError if no model is provided and the trainer doesn't have a reference to the model. What do you think ?

This is what's done in #7605

**Comment by zmurez** [comment]
Thanks all!
